File: p_009.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        else {
            int l = 0;
            while (x/int(pow(10, l)) >= 10) {
                l++;
            }
            for (int i = l; i > l / 2; i--) {
                int a = x/int(pow(10, i))%10;
                int b = x%int(pow(10, (l-i+1)))/(pow(10, (l-i)));
                if (a != b) {
                    return false;
                }
            }
            return true;
        }
    }
};

File: python/p_009.py
class Solution:
    # @param {integer} x
    # @return {boolean}
    def isPalindrome(self, x):
        if x < 0:
            return False
        else:
            # get length
            l = 0
            while x/(10**l) >= 10:
                l += 1
            # print l
            for i in range(l, l/2, -1):
                # print x/(10**i)%10, x%(10**(l-i+1))/(10**(l-i))
                if x/(10**i)%10 != x%(10**(l-i+1))/(10**(l-i)):
                    return False
            return True
